crates/core/src/execution.rs

```diff
@@ -1,10 +1,19 @@
-use anyhow::Result;
+use std::env;
+
+use anyhow::{bail, Result};
 use quickjs_wasm_rs::Context;
 
 pub fn run_bytecode(context: &Context, bytecode: &[u8]) -> Result<()> {
     context.eval_binary(bytecode)?;
     if cfg!(feature = "experimental_event_loop") {
         context.execute_pending()?;
+    } else if context.is_pending() && !is_running_wpt_suite() {
+        // WPT enqueues a promise as part of its suite setup and we can't change that behaviour so don't error
+        bail!("Adding tasks to the event queue is not supported");
     }
     Ok(())
 }
+
+fn is_running_wpt_suite() -> bool {
+    env::var("JAVY_WPT").is_ok()
+}
```

Comment:
What are we hoping to do with the `experimental_event_loop` feature? Is it experimental just for testing for now, but do we eventually want to promote it to the default behaviour? If so, I feel like running the WPT suite with it enabled should be fine since that's where we want to be eventually, but I could be missing something in this assumption

Reply:
> I want to run the WPT suite with Javy in state (2) because (1) potentially changes Javy's behaviour under test and (3) causes a panic on suite setup. Unfortunately it's a pain to figure out how to make that work without some sort of compromise somewhere.

Isn't 3 the "correct" behaviour? I'd argue that if WPT is enqueuing jobs, we should be providing the right behaviour at the core layer and not something that we consider "right". That's my main reason to suggest enabling the event loop. Also, can you provide a concrete example for:

> If we use a build of Javy with the experimental event loop flag enabled, what we're testing is something that potentially behaves differently in material ways. The tests could have different results than if it were disabled. I'd prefer not running the event loop because of that.
 
I'm failing to see how this could be an issue given that we are relying on the event loop for our own tests: we are not preventing new jobs from getting added, it's just that we are not flushing the queue.

-- 

As a bottom line:

(i) we are changing the behaviour here by saying: "if we are running WPT tests, we don't check for pending jobs", which is non-standard afaik, so this warrants a different build IMO: instead of using an env var, we might as well introduce a new feature, such as `wpt`. I'd be more comfortable with that. I do worry that the complexity of build types is going up, but I think it's less error prone and more explicit than an env var.

(ii) Given that our tests rely on the queue, I'd argue that getting the right level of compatibility for the event loop should be prioritized. Without a concrete example of how the event loop can introduce different results in our tests, I still think that enabling the event loop in the tests is better than introducing custom behaviour.